File: Src/Icm.ContextConsole/Application/IApplication.vb
Imports Icm.Localization

Public Interface IApplication

    Property CurrentContextNode As ITreeNode(Of IContext)
    Property RootContextNode As ITreeNode(Of IContext)

    Function GetAllContexts() As IEnumerable(Of IContext)

    Function ExecuteCommand() As Boolean

    Property TokenQueue As Queue(Of String)

    Property Interactor As IInteractor

    Property InternalLocRepo As ILocalizationRepository
    Property ExternalLocRepo As ILocalizationRepository

End Interface

File: Src/Icm.ContextConsole/Application/IApplication.cs
using System.Collections.Generic;
using Icm.Localization;
using Icm.Tree;

public interface IApplication
{


	string ApplicationPrompt { get; set; }
	ITreeNode<IContext> CurrentContextNode { get; }

	ITreeNode<IContext> RootContextNode { get; }
	IEnumerable<IContext> GetAllContexts();

	bool ExecuteCommand();


	IInteractor Interactor { get; }
	ILocalizationRepository InternalLocRepo { get; }

	ILocalizationRepository ExternalLocRepo { get; }

	void Run();

	IDependencyResolver DependencyResolver { get; set; }
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
